core/src/main/java/com/github/jeanbaptistewatenberg/junit5kubernetes/core/Service.java

```
@@ -0,0 +1,166 @@
+                client.setDebugging(true);
+            }
+            // infinite timeout
+            OkHttpClient.Builder builder = client.getHttpClient().newBuilder()
+                    .readTimeout(0, TimeUnit.SECONDS);
+
+            if (DISABLE_HTTP2 != null && DISABLE_HTTP2.equalsIgnoreCase("true")) {
+                builder.protocols(Collections.singletonList(Protocol.HTTP_1_1));
+            }
+
+            client.setVerifyingSsl(false);
+
+            OkHttpClient httpClient = builder.build();
+            client.setHttpClient(httpClient);
+            Configuration.setDefaultApiClient(client);
+
+            coreV1Api = new CoreV1Api(client);
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return coreV1Api;
+    }
+
+    @Override
+    public String getObjectName() {
+        V1Service v1Service = this.createdService.get();
+        if (v1Service != null) {
+            return v1Service.getMetadata().getName();
+        } else {
+            throw new RuntimeException("Can't get name of a non running object.");
+        }
+    }
+
+    @Override
+    public String getObjectHostIp() {
+        V1Service v1Service = this.createdService.get();
+        if (v1Service != null) {
+            try {
+                V1Service retrievedService = coreV1Api.readNamespacedService(v1Service.getMetadata().getName(), NAMESPACE, null, null, null);
+                if (retrievedService.getStatus() == null) {
+                    throw new RuntimeException("Can't get ip of a non running object.");
+                }
+                //Anhand der Umgebungsvariable entscheiden, ob der Test im Cluster oder lokal ausgeführt wird
+                if(System.getenv("KUBERNETES_PORT") != null) {
+                    return retrievedService.getSpec().getClusterIP();
+                }else{
+                    return retrievedService.getSpec().getExternalIPs().get(0);
```

In some clusters a service may not have any `externalIps` and under those circumstances I think we should raise an exception with an explicit message.